transactions.go

```diff
@@ -361,6 +382,91 @@ func (lw *LibWallet) DecodeTransaction(txHash []byte) (string, error) {
 	return string(result), nil
 }
 
+func (lw *LibWallet) FilterTransactions(txFilter int32) (string, error) {
+	query := lw.prepareTxQuery(txFilter)
+	var transactions []Transaction
+
+	err := query.Find(&transactions)
+	if err != nil {
+		return "", nil
+	}
+
+	jsonEncodedTransactions, err := json.Marshal(&transactions)
+	if err != nil {
+		return "", err
+	}
+
+	return string(jsonEncodedTransactions), nil
+}
+
+func (lw *LibWallet) CountTransactions(txFilter int32) (int, error) {
+	query := lw.prepareTxQuery(txFilter)
+
+	count, err := query.Count(&Transaction{})
+	if err != nil {
+		return -1, err
+	}
+
+	return count, nil
+}
+
+func (lw *LibWallet) DetermineTxFilter(txType string, txDirection int32) int32 {
+
+	if txType != TxTypeRegular {
```

Comment:
These nested if statements can be simplified thus:
```
if txType == TxTypeCoinBase {
	return TxFilterCoinBase
}
if txType != TxTypeRegular {
	return TxFilterStaking
}
```

Also, the empty line at the start of the function can be done without.